src/autoschedulers/anderson2021/FunctionDAG.cpp

```diff
@@ -0,0 +1,1298 @@
+            if (e->producer->func.name() == name) {
+                // The same name can be encountered multiple times
+                // (e.g. a+a, where a is a trivial function),
+                // so we can't use std::move(matrix) here without making a copy
+                e->add_load_jacobian(matrix);
+            }
+        }
+    }
+
+public:
+    Featurizer(Function &func, FunctionDAG::Node::Stage &stage)
+        : func(func), stage(stage) {
+    }
+
+    void visit_store_args(const std::string &name, Type t, vector<Expr> args) {
+        for (auto &e : args) {
+            e = common_subexpression_elimination(simplify(e));  // Get things into canonical form
+        }
+        visit_memory_access(name, t, args, PipelineFeatures::AccessType::Store);
+    }
+};
+
+}  // namespace
+
+void LoadJacobian::dump(const char *prefix) const {
+    if (count() > 1) {
+        aslog(0) << prefix << count() << " x\n";
+    }
+    for (size_t i = 0; i < producer_storage_dims(); i++) {
+        aslog(0) << prefix << "  [";
+
+        for (size_t j = 0; j < consumer_loop_dims(); j++) {
+            const auto &c = (*this)(i, j);
+            if (!c.exists()) {
+                aslog(0) << " _  ";
+            } else if (c.denominator == 1) {
+                aslog(0) << " " << c.numerator << "  ";
+            } else {
+                aslog(0) << c.numerator << "/" << c.denominator << " ";
+            }
+        }
+        aslog(0) << "]\n";
+    }
+    aslog(0) << "\n";
+}
+
+void BoundContents::validate() const {
+    for (int i = 0; i < layout->total_size; i++) {
+        auto p = data()[i];
+        if (p.max() < p.min()) {
+            aslog(0) << "Bad bounds object:\n";
+            for (int j = 0; j < layout->total_size; j++) {
+                if (i == j) {
+                    aslog(0) << "=> ";
+                } else {
+                    aslog(0) << "   ";
+                }
+                aslog(0) << j << ": " << data()[j].min() << ", " << data()[j].max() << "\n";
+            }
+            internal_error << "Aborting";
+        }
+    }
+}
+
+BoundContents::Layout::~Layout() {
+    internal_assert(num_live == 0)
+        << "Destroying a Layout without returning all the BoundContents. "
+        << num_live << " are still live\n";
+    for (auto *b : pool) {
+        b->~BoundContents();
+    }
+    for (auto *b : blocks) {
+        free(b);
+    }
+}
+
+void BoundContents::Layout::allocate_some_more() const {
+    size_t size_of_one = sizeof(BoundContents) + total_size * sizeof(Span);
+    const size_t number_per_block = std::max((size_t)8, 4096 / size_of_one);  // Make a page of them, or 8, whichever is larger.
+    const size_t bytes_to_allocate = std::max(size_of_one * number_per_block, (size_t)4096);
+    unsigned char *mem = (unsigned char *)malloc(bytes_to_allocate);
+
+    blocks.push_back(mem);
+    static_assert((sizeof(BoundContents) & 7) == 0, "BoundContents header is not aligned");
```

Comment:
Are you trying to ensure that this is aligned to size of a pointer? If so, do that explicitly: `(sizeof(BoundContents) % sizeof(void*)) == 0`

EDIT: I see that the same pattern is in the adams2019 version too, but it is also naughty, we'll fix it later